### PR TITLE
adds `start_command` to service resource

### PR DESCRIPTION
**Pull request by matifali**
This is my first time working on a terraform provider and go. I did not add the tests. I would be happy if you could guide me in completing it.

### CODE REVIEW

**Comment by pksunkara** [comment]
`startCommand` is available in railway config file. https://docs.railway.app/deploy/config-as-code

As mentioned in the documentation:

> ⚠️ NOTE: All the other settings not specified here are recommended to be specified in the Railway config file.

Therefore, I am rejecting this PR.

**Comment by matifali** [comment]
@pksunkara I am trying to make it closer or other terraform providers that directly support a start command. For example fly.io. It will make it easier to just specify an image source and start command without needing to create a railway config file and track it in a repo. 

```hcl
resource railway_service "workspace" {
  image_source = "codercom/enterprise-base"
  start_command = coder_agent.me.init_script
}
```
The issue with a railway config is that it is not straight forward to use a dynamic startup script as the start command. 

I am doing this in context of creating a native https://github.com/coder/coder template to be used with https://railway.app/template/coder. 

Thanks.

**Review by pksunkara — CHANGES_REQUESTED** [review]

**Comment by pksunkara** [comment]
> I did not add the tests. I would be happy if you could guide me in completing it.

https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests is a good read.

File: internal/provider/resource_service.go [hunk 1]
```diff
@@ -106,6 +107,13 @@ func (r *ServiceResource) Schema(ctx context.Context, req resource.SchemaRequest
 					stringvalidator.UTF8LengthAtLeast(1),
 				},
 			},
+			"start_command": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (matifali) [reply internal/provider/resource_service.go:110]: Sure, I will do that.

**Comment by pksunkara** [comment]
Closing this until further update.

**Comment by matifali** [comment]
I don't think I will be able to find time to finish this any time in the near future. Thank you!

**Pull request closed without merging**